pkg/job/job.go

```diff
@@ -305,129 +308,33 @@ func (k *kubeJob) GenerateISO(ctx context.Context, cluster common.Cluster, jobNa
-									Name:  "S3_BUCKET",
-									Value: k.Config.S3Bucket,
-								},
-								{
-									Name:  "S3_REGION",
-									Value: k.Config.S3Region,
-								},
-								{
-									Name:  "CLUSTER_ID",
-									Value: id.String(),
-								},
-								{
-									Name:  "OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE",
-									Value: k.ReleaseImage, //TODO: change this to match the cluster openshift version
-								},
-								{
-									Name:  "AWS_ACCESS_KEY_ID",
-									Value: k.Config.AwsAccessKeyID,
-								},
-								{
-									Name:  "AWS_SECRET_ACCESS_KEY",
-									Value: k.Config.AwsSecretAccessKey,
-								},
-								{
-									Name:  "SKIP_CERT_VERIFICATION",
-									Value: strconv.FormatBool(k.Config.SkipCertVerification),
-								},
-							},
-							Resources: core.ResourceRequirements{
-								Limits: core.ResourceList{
-									"cpu":    getQuantity(k.Config.JobCPULimit),
-									"memory": getQuantity(k.Config.JobMemoryLimit),
-								},
-								Requests: core.ResourceList{
-									"cpu":    getQuantity(k.Config.JobCPURequests),
-									"memory": getQuantity(k.Config.JobMemoryRequests),
-								},
-							},
-						},
-					},
-					RestartPolicy: "Never",
-				},
-			},
-		},
-	}
-	if encodedDhcpFileContents != "" {
-		ret.Spec.Template.Spec.Containers[0].Env = append(ret.Spec.Template.Spec.Containers[0].Env,
-			core.EnvVar{
-				Name:  "DHCP_ALLOCATION_FILE",
-				Value: encodedDhcpFileContents,
-			})
-	}
-	return ret
-}
-
-// creates install config
+// GenerateInstallConfig creates install config and ignition files
 func (k *kubeJob) GenerateInstallConfig(ctx context.Context, cluster common.Cluster, cfg []byte) error {
 	log := logutil.FromContext(ctx, k.log)
-	ctime := time.Time(cluster.CreatedAt)
-	cTimestamp := strconv.FormatInt(ctime.Unix(), 10)
-	jobName := fmt.Sprintf("%s-%s-%s", ignitionGeneratorPrefix, cluster.ID.String(), cTimestamp)[:63]
-	encodedDhcpFileContents, err := network.GetEncodedDhcpParamFileContents(&cluster)
-	if err != nil {
-		wrapped := errors.Wrapf(err, "Could not create DHCP encoded file")
-		log.WithError(wrapped).Errorf("GenerateInstallConfig")
-		return wrapped
+	workDir := filepath.Join(k.Config.WorkDir, cluster.ID.String())
+	err := os.Mkdir(workDir, 0755)
+	if err != nil && !os.IsExist(err) {
+		return err
 	}
-	if err := k.Create(ctx, k.createKubeconfigJob(&cluster, jobName, cfg, encodedDhcpFileContents)); err != nil {
-		log.WithError(err).Errorf("Failed to create kubeconfig generation job %s for cluster %s", jobName, cluster.ID)
-		return errors.Wrapf(err, "Failed to create kubeconfig generation job %s for cluster %s", jobName, cluster.ID)
+	defer func() {
+		err := os.RemoveAll(workDir)
+		if err != nil {
+			log.WithError(err).Error("Failed to clean up generated ignition directory")
+		}
+	}()
+
+	// runs openshift-install to generate ignition files, then modifies them as necessary
+	err = ignition.Generate(workDir, cfg, &cluster, k.Config.ReleaseImage, log)
```

Comment:
The Generate function already adds `cluster_id` to the path before creating and manipulating files. That's important so that the files are in the right place to be served by the non-S3 storage implementation from #151.

Two calls in parallel for the same cluster would overwrite each other's files with the same content, so I don't think there would be any harm. We could add locking at some layer, or run this in a single goroutine to force multiple requests to be handled in serial. But I'm not sure that's worth doing.